app/src/main/java/com/celzero/bravedns/util/Utilities.kt

```diff
@@ -508,6 +508,23 @@ class Utilities {
             return ctx.filesDir.canonicalPath + File.separator + timestamp + File.separator + which
         }
 
+        fun deleteUnwantedLocalBlocklistFolders(fileOrDirectory: File,
+                                                currentActiveFolder: String) {
+            // folders with timestamp other than current local blocklist timestamp
+            // will be deleted (path: ../files/<timestamp>)
+            if (fileOrDirectory.isDirectory) {
```

Comment:
This `if` also needs

```
!fileOrDirectory.name.equals(currentActiveFolder)
```

Better yet, pass `skip` as a fn and use it as a guard.

```
fun deleteUnwanted.. (fileOrDir: File, fun skip (String) -> Boolean). {
    ...
    if (skip(filename)) continue
    
    if ...
    if ...
}
```

Reply:
Modified as below
```
fun cleanupOldLocalBlocklistFolders(fileOrDirectory: File, currentActiveFolder: String) {
   if (fileOrDirectory.name.equals(currentActiveFolder)) return

   if (fileOrDirectory.name.startsWith("16")) {
        deleteRecursive(fileOrDirectory)
   }
            
   if (fileOrDirectory.isDirectory) {
        fileOrDirectory.listFiles()?.forEach { child ->
           cleanupOldLocalBlocklistFolders(child, currentActiveFolder)
        }
   }
}
```